resources/session02/homework/http_server.py

```diff
@@ -35,7 +42,37 @@ def parse_request(request):
 
 def resolve_uri(uri):
     """This method should return appropriate content and a mime type"""
-    return b"still broken", b"text/plain"
+    contents = b''
+    mimetype = b''
+    directory = './webroot'
+
+    if isinstance(uri, bytes):
+        directory += str(uri.decode('utf8'))
+    else:
+        directory += str(uri)
+
+    try:
+        pathreq = Path(directory).resolve()  # Attempt to resolve the requested directory
+
+        if pathreq.exists():
+            if pathreq.is_dir():
+                # The requested path maps to a directory. The content should be a
+                # plain-text listing of the directory contents with a mimetype of text/plain.
+                filelist = [str(files) for files in pathreq.iterdir() if files.is_file()]
+                contents = '\n'.join(filelist).encode('utf8')
+                mimetype = b'text/plain'
+            else:
+                # The requested path maps to a file
+                contents = pathreq.read_bytes()
+                mimetype = mimetypes.guess_type(str(pathreq))[0].encode('utf8')
+        else:
+            raise NameError("{} does not exist\n".format(pathreq))
+    except FileNotFoundError:
+        raise NameError("{} could not be resolved\n".format(directory))
+    except:
+        print("Unexpected error:{}\n".format(sys.exc_info()[0]))
```

Comment:
You are swallowing exceptions here.  Never a good idea.  unexpected exceptions should be allowed to raise and handled from outside this function.